File: checkio_js/home/python/popularWords.py
import re

def popularWords(text, words):
    num = [len(re.findall(re.compile(f'\\b{xxx}\\b', re.I), text)) for xxx in words]
    d = {}
    for x, y in zip (words, num):
        d[x] = y
    return d


if __name__ == '__main__':
    print('Example:')
    print(popularWords('''
When I was One
I had just begun
When I was Two
I was nearly new''', ['i', 'was', 'three', 'near']))

assert popularWords('When I was One I had just begun When I was Two I was nearly new',
                        ['i', 'was', 'three', 'near']) == {'i': 4, 'was': 3, 'three': 0, 'near': 0}

File: checkio_js/home/javascript/popularWords.js
"use strict";

function popularWords(text, words) {
    return words.reduce((k,v) => (k[v] = text.match(RegExp(`(\\b${v}\\b)`, "gi"))?.length || 0, k), {})
    // let popular = {}
    /**
    words.map(a => popular[a] = text.match(RegExp(`\\b${a}\\b`, 'gi')) == null ?
        0 : text.match(new RegExp(`\\b${a}\\b`, 'gi')).length)
    **/
    // words.map(a => popular[a] = text.match(RegExp(`\\b${a}\\b`, 'gi'))?.length || 0)
    // return popular
}
// return words.reduce((k,v) => (k[v] = text.match(RegExp(`(\\b${v}\\b)`, "gi"))?.length || 0, k), {})

console.log('Example:')
console.log(popularWords(`
When I was One
I had just begun
When I was Two
I was nearly new`, ['i', 'was', 'three', 'near']))
